src/Bridge/Laravel/Providers/RequestServiceProvider.php

```diff
@@ -28,10 +28,33 @@ public function register(): void
             // Set proxy list
             HttpRequest::setTrustedProxies(
                 \explode(',', $env->get('TRUSTED_PROXIES') ?? ''),
-                HttpRequest::HEADER_X_FORWARDED_ALL
+                $this->mapTrustedHeader($env->get('TRUSTED_PROXIES_HEADER') ?? '')
             );
 
             return $this->app->make(Request::class);
         });
     }
+
+    /**
+     * Maps the trusted header string value to integer.
+     *
+     * @param string $value
+     *
+     * @return int
+     */
+    private function mapTrustedHeader(string $value): int
+    {
+        $constant = \constant(
+            \sprintf('%s::%s', HttpRequest::class, $value)
+        );
+
+        if ($constant !== null) {
+            return $constant;
```

Comment:
The return type of this method is type-hinted as `int`, but the constant function returns `mixed`
There should probably be some type checking before return to ensure a `TypeError` does cannot occur

Reply:
also, we should probably check the constants existence via `defined()` before calling `constant()` to avoid the PHP Warning in that event, which could supersede the exception being thrown below depending on the exception/error handling of the implementing application

Reply:
can you use `defined` against class constants?

Reply:
also, yes about the int type checking - what was I thinking!?

Reply:
```
Psy Shell v0.9.9 (PHP 7.3.12 — cli) by Justin Hileman
>>> namespace BigStrongTest;
>>> class NiceOne {
...     public const SOMETHING_GOOD = 1;
... }
>>> \defined('BigStrongTest\NiceOne::SOMETHING_GOOD');
=> true
```

Reply:
old mate psy shell